contracts/core/OTCWrapperV2.sol

```diff
@@ -0,0 +1,1046 @@
+            0, // not used
+            "" // not used
+        );
+
+        // execute actions
+        controller.operate(actions);
+
+        emit VaultSettled(_orderID);
+    }
+
+    /**
+     * @notice allows user to redeem after expiry
+     * @dev can only be called by the user who is the order buyer
+     * @param _orderID order id
+     */
+    function redeem(uint256 _orderID) external nonReentrant {
+        require(orderStatus[_orderID] == OrderStatus.Succeeded, "OTCWrapper: inexistent or unsuccessful order");
+
+        Order memory order = orders[_orderID];
+
+        require(order.buyer == _msgSender(), "OTCWrapper: sender is not the order buyer");
+
+        // check if vault has enough collateral to cover payout
+        (UtilsWrapperInterface.Vault memory vault, uint256 typeVault, ) = controller.getVaultWithDetails(
+            address(this),
+            order.vaultID
+        );
+
+        (, bool isValidVault) = calculator.getExcessCollateral(vault, typeVault);
+
+        // require that vault is valid (has excess collateral) before redeeming
+        // to avoid allowing redeeming undercollateralized vaults
+        require(isValidVault, "OTCWrapper: insuficient collateral to redeem");
+
+        UtilsWrapperInterface.ActionArgs[] memory actions = new UtilsWrapperInterface.ActionArgs[](1);
+
+        actions[0] = UtilsWrapperInterface.ActionArgs(
+            UtilsWrapperInterface.ActionType.Redeem,
+            address(0), // not used
+            order.buyer, // address to transfer to
+            order.oToken, // otoken address
+            0, // not used
+            order.size, // otoken amount
+            0, // not used
+            "" // not used
+        );
+
+        // execute actions
+        controller.operate(actions);
+
+        orders[_orderID].size = 0;
+
+        emit Reedem(_orderID, order.size);
+    }
+
+    /**
+     * @notice sends fees to beneficiary
+     */
+    function claimFees() external {
+        IERC20 usdc = IERC20(USDC);
+
+        uint256 amountToClaim = usdc.balanceOf(address(this));
+        require(amountToClaim > 0, "OTCWrapper: amount to claim cannot be 0");
+
+        usdc.safeTransfer(beneficiary, amountToClaim);
+
+        emit FeesClaimed(amountToClaim);
+    }
+
+    /**
+     * @notice Allows current buyer to unwind the position
+     * @dev can only be called by whitelisted market makers
+     * @param _sellerOrderSignature seller order unwind signature
+     * @param _bidderOrderSignature bidder order unwind signature
+     * @param _bidderUSDCSignature bidder permit signature
+     */
+    function sellRedeemRights(
+        UnwindPermit calldata _sellerOrderSignature,
+        UnwindPermit calldata _bidderOrderSignature,
+        Permit calldata _bidderUSDCSignature
+    ) external nonReentrant {
+        // ensure order signature integrity
+        UNWIND_PERMIT.checkOrderPermit(
+            _sellerOrderSignature.acct,
+            _sellerOrderSignature.orderID,
```

Comment:
actually wdyt about stacking the orderID into the `uint256 value` integer, say in the first 32 bits and keep the actual bid value in the other 224 bits. Since order id just increases incrementally shouldn't really be possible to overflow in our lifetime.